## 0. 1주차 - 기초 통계 과제는 회귀분석을 직접 해보는 것입니다.
## 1번에서는 회귀분석 결과에 대한 해석을 서술형으로 작성해주시고,
## 2번에서는 직접 코드를 작성까지해주세요.

## 1. 회귀분석 결과 해석

In [2]:
import pandas as pd
url = "https://www.statlearning.com/s/Advertising.csv"
df1 = pd.read_csv(url)
df1.head()

Unnamed: 0     TV  radio  newspaper  sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

1번 문제에서 사용할 데이터는  'Advertising' 데이터 셋으로 3개의 매체에 투입한 광고비와 해당 제품의 판매량을 나타내고 있습니다.

1. TV : TV 광고비 (천 달러 단위)
2. Radio : 라디오 광고비
3. Newspaper : 신문 광고비
4. Sales : 제품 판매량 (단위: 천 개)

아래의 회귀분석 코드를 실행하시고, 결과 해석을 적어주세요.

In [3]:
import statsmodels.api as sm
from sklearn.linear_model import LinearRegression

X = df1[["TV"]]
y = df1["sales"]
X = sm.add_constant(X)
model = sm.OLS(y, X).fit()

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                  sales   R-squared:                       0.612
Model:                            OLS   Adj. R-squared:                  0.610
Method:                 Least Squares   F-statistic:                     312.1
Date:                Tue, 22 Jul 2025   Prob (F-statistic):           1.47e-42
Time:                        06:03:38   Log-Likelihood:                -519.05
No. Observations:                 200   AIC:                             1042.
Df Residuals:                     198   BIC:                             1049.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          7.0326      0.458     15.360      0.0

# 답변 :

### ex. 평균적으로 TV 광고비에 천 달러를 더 투자하면 제품 판매량이 ~~개 증가하며, 이 효과는 통계적으로 (유의하다 / 유의하지 않다).

종속변수는 sales, 독립변수는 TV이므로 회귀식은 sales = 7.0326 + 0.0475 * TV 가 될 것이다.\
따라서, 평균적으로 TV 광고비에 천 달러를 더 투자하면 제품 판매량이 47.5만큼 증가한다.\
두 계수 모두 p-value가 0.000 이므로 통계적으로 유의하다.'''

##2. 회귀분석 코드 작성 및 해석

In [4]:
from sklearn.datasets import fetch_california_housing
import pandas as pd

housing = fetch_california_housing(as_frame=True)
df2 = housing.frame

print(df2.head())

   MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  Latitude  \
0  8.3252      41.0  6.984127   1.023810       322.0  2.555556     37.88   
1  8.3014      21.0  6.238137   0.971880      2401.0  2.109842     37.86   
2  7.2574      52.0  8.288136   1.073446       496.0  2.802260     37.85   
3  5.6431      52.0  5.817352   1.073059       558.0  2.547945     37.85   
4  3.8462      52.0  6.281853   1.081081       565.0  2.181467     37.85   

   Longitude  MedHouseVal  
0    -122.23        4.526  
1    -122.22        3.585  
2    -122.24        3.521  
3    -122.25        3.413  
4    -122.25        3.422  


이번 실습에서는 캘리포니아 주택 가격(California Housing) 데이터를 사용합니다.
이 데이터는 1990년 인구조사 데이터를 바탕으로 지역별 주택 가격에 영향을 줄 수 있는 요인들을 포함하고 있습니다.

1. MedInc :	지역의 중위소득 (단위: 만 달러)
2. HouseAge :	해당 지역 주택들의 평균 연식 (단위: 년)
3. AveRooms :	주택 수 대비 총 방 수 비율
4. AveBedrms	: 주택 수 대비 총 침실 수 비율
5. Population : 	지역의 총 인구 수
6. AveOccup : 총 인구 / 가구 수
7. Latitude :	위도 정보
8. Longitude :	경도 정보
9. MedHouseVal :	중위 주택 가격 (단위: 십만 달러)

MedHouseVal를 종속변수로, MedInc을 독립변수로 하는 단일회귀분석 코드를 완성해주세요.

In [7]:
X = df2[["MedInc"]]
y = df2["MedHouseVal"]
X = sm.add_constant(X)
model = sm.OLS(y, X).fit()

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:            MedHouseVal   R-squared:                       0.473
Model:                            OLS   Adj. R-squared:                  0.473
Method:                 Least Squares   F-statistic:                 1.856e+04
Date:                Tue, 22 Jul 2025   Prob (F-statistic):               0.00
Time:                        06:09:53   Log-Likelihood:                -25623.
No. Observations:               20640   AIC:                         5.125e+04
Df Residuals:                   20638   BIC:                         5.127e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.4509      0.013     34.082      0.0